frontend-ssr/pages/api/user/login.js

```diff
@@ -0,0 +1,38 @@
+//Imports
+const User = require("../../../models/User.schema");
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+const { loginValidation } = require("../../../utils/validation/userValidation")
+
+//Vars
+const error_message = { status: "fail", message: 'Email or Password is incorrect' }
+
+export default async (req, res) => {
+    let { method, body } = { ...req };
+    switch (method) {
+        case "POST":
+            //destruct the req
+            let { email, password } = { ...body };
+            let { error } = loginValidation({ email, password })
+            //Check valid form 
+            if (error) return res.status(400).json(error_message);
+            //Check user base on username or email
+            let user = await User.findOne({ $or: [{ username: email }, { email: email }] });
+            if (!user) return res.status(400).json(error_message);
+            //User exists => Check password  
+            const validPass = await bcrypt.compare(password, user.password);
+            if (!validPass) return res.status(400).json(error_message);
```

Comment:
Let's give more specific error messages. If username or email doesn't exist when logging it, let's say that. Otherwise, if the password is wrong let's give a specific message regarding this.